plonky2x/src/frontend/eth/storage/generators/storage.rs

```diff
@@ -118,3 +120,114 @@ impl<F: RichField + Extendable<D>, const D: usize> SimpleGenerator<F, D>
         })
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct EthLogGenerator<F: RichField + Extendable<D>, const D: usize> {
+    transaction_hash: Bytes32Variable,
+    block_hash: Bytes32Variable,
+    log_index: u64,
+    pub value: EthLogVariable,
+    chain_id: u64,
+    _phantom: PhantomData<F>,
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> EthLogGenerator<F, D> {
+    pub fn new(
+        builder: &mut CircuitBuilder<F, D>,
+        transaction_hash: Bytes32Variable,
+        block_hash: Bytes32Variable,
+        log_index: u64,
+    ) -> EthLogGenerator<F, D> {
+        let chain_id = builder.get_chain_id();
+        let value = builder.init::<EthLogVariable>();
+        EthLogGenerator {
+            transaction_hash,
+            block_hash,
+            log_index,
+            value,
+            chain_id,
+            _phantom: PhantomData::<F>,
+        }
+    }
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> SimpleGenerator<F, D> for EthLogGenerator<F, D> {
+    fn id(&self) -> String {
+        "EthLogGenerator".to_string()
+    }
+
+    fn dependencies(&self) -> Vec<Target> {
+        let mut targets = Vec::new();
+        targets.extend(self.transaction_hash.targets());
+        targets.extend(self.block_hash.targets());
+        targets
+    }
+
+    fn run_once(&self, witness: &PartitionWitness<F>, buffer: &mut GeneratedValues<F>) {
+        let transaction_hash = self.transaction_hash.get(witness);
+        // block_hash is unused
+        let _block_hash = self.block_hash.get(witness);
+
+        let provider = get_provider(self.chain_id);
+        let rt = Runtime::new().expect("failed to create tokio runtime");
+        let result: TransactionReceipt = rt
+            .block_on(async {
+                provider
+                    .get_transaction_receipt(transaction_hash)
+                    .await
+                    .expect("Failed to call get_transaction_receipt")
+            })
+            .expect("No transaction receipt found");
+
+        let log = &result.logs[self.log_index as usize];
+        let value = EthLog {
+            address: log.address,
+            topics: [log.topics[0], log.topics[1], log.topics[2]],
+            data_hash: ethers::types::H256::from_slice(sha2::Sha256::digest(&log.data).as_ref()),
```

Comment:
very small nit: let's just import `H256` and `Sha256` directly from the crates instead of having these long names